pulp_ostree/app/tasks/uploading.py

```diff
@@ -0,0 +1,263 @@
+import glob
+import os
+import tarfile
+import uuid
+
+from gettext import gettext
+
+from pulpcore.plugin.models import Artifact, Repository, ProgressReport
+from pulpcore.plugin.stages import (
+    ArtifactSaver,
+    ContentSaver,
+    DeclarativeVersion,
+    DeclarativeArtifact,
+    DeclarativeContent,
+    QueryExistingArtifacts,
+    QueryExistingContents,
+    Stage,
+)
+
+from pulp_ostree.app.models import (
+    OstreeRefsHead,
+    OstreeCommit,
+    OstreeConfig,
+    OstreeObject,
+    OstreeObjectType,
+)
+
+import gi
+
+gi.require_version("OSTree", "1.0")
+from gi.repository import Gio, GLib, OSTree  # noqa: E402: module level not at top of file
+
+
+def parse_tarball(artifact_pk, repository_pk, repository_name):
+    """Upload content to an OSTree repository.
+
+    Args:
+        artifact_pk (str): The PK of an artifact that identifies an uploaded tarball.
+        repository_pk (str): The repository PK.
+        repository_name (str): The name of an OSTree repository (e.g., "repo")
+
+    Raises:
+        ValueError: If an OSTree repository could not be properly parsed.
+    """
+    tarball_artifact = Artifact.objects.get(pk=artifact_pk)
```

Comment:
I am not sure what kind of upload serializers you are referring to. Having a large file, we are supposed to upload it in chunks and commit it. Committing the upload results in creating an artifact. If not committed, we have a bunch of chunks residing on the disk referenced by an `Upload` object (https://github.com/pulp/pulpcore/blob/acb23f9280e2c33fde29d91bb3b7a627b85f206f/pulpcore/app/tasks/upload.py#L28-L39). Would adding a new function for only merging uploaded chunks to pulpcore and using it be better?

@dkliban and I discussed this and we came to the conclusion that the current approach ((1) pulpcore part - upload a tarball, create an artifact, (2) pulp_ostree part - parse artifact, create a new OSTree repository version, remove the artifact) is satisfactory.

Reply:
I agree that this is the only way without adding features to pulpcore.
But it would be interesting in the future, if we could specify upload objects instead of artifacts or uploaded files for content creation.